bomb_lab.md

```diff
@@ -0,0 +1,225 @@
+---
+layout: doc
+title: "Bomb Lab"
+learning_objectives:
+  - Getting familiar with GDB and system programming.
+wikibook:
+  - "HW0"
+  - "System Programming Jokes"
+---
+
+## HW0
+
+You have already been assigned a HW0 for the class. [Here is the HW0 submission link](https://goo.gl/forms/8yX0n9E5CTkSeg4v2) -- beware the link does not save so be ready to submit when going through. We'll spend the first part of class going over the various questions from HW0. Your grade for this lab is partly HW0 and partly the assignment below.
+
+## Development
+
+Read the [Development Guide](./development.html)!
+
+## Git
+
+(Read the first section of the development site!) You will be using `git` to submit all your assignments in this course. First go to [the repository creator](https://edu.cs.illinois.edu/create-ghe-repo/{{site.subject_code}}{{site.course_number}}-{{site.semester}}/). **DO NOT DO THE README TUTORIAL**
+
+**Once you are in your VM**, you'll need to set up some global defaults
+
+```console
+git config --global user.name "FIRST_NAME LAST_NAME"
+git config --global user.email NETID@example.com
+```
+
+Make sure to replace `FIRST_NAME` and `LAST_NAME` and `NETID` with your information. Your code will not be graded if you git config is not correctly set. Then checkout your repository as follows
+
+```console
+git clone https://github-dev.cs.illinois.edu/{{ site.subject_code }}{{site.course_number}}-{{site.semester }}/NETID.git
+```
+
+which will check out your entire git repo into a folder called 'NETID' into your current directory. Now change your directory into that folder:
+
+```console
+cd NETID
+```
+
+You've probably noticed the repository is empty! In order to grab the latest version of our assignment, complete the following steps.
+
+```console
+git remote add release https://github-dev.cs.illinois.edu/{{ site.subject_code }}{{site.course_number}}-{{site.semester }}/_release.git
+git pull release master
+git push origin master
+```
+
+## Graded Files
+
+One section we will have on the top of every assignment is a section called `graded_files` these are the files that we use to grade the assignment. `git` does not allow us to set the rest of the files to readonly to prevent changing them (header files, Makefile). You have to be careful to avoid changing these files. We are working on a system that prevents you from accidentally committing the changes, but there is no easy way to set permissions on your local copy.
+
+## Background and Assignment
+
+You are working for ShadyCorp Inc. Your boss discover's that your competitor, ShadierCorp Inc., has uploaded a bomb to your company's only production server. You need to safely defuse the bomb. Luckily you boss has found a defuser to help you.
```

Comment:
maybe let’s make it clear this is an entirely fictional acenario so no one is worried